### PR TITLE
Improve docs on Supported Resources

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Makes the output location of the list of supported resources configurable, so we can place it directly in the right place for our documentation site

**Special notes for your reviewer**:

Moves the introductory text into our Yaml file to make it easier to update

Preserves any preexisting frontmatter in the output file, allowing us to tweak the website without requiring code changes to the generator

Creates a separate object containing configuration for the report, partly to keep things nicely encapsulated, and also as prepositioning for a solution to #2337 

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: hack/crossplane/apis/resources.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 ---
 title: Supported Resources
 ---
+These are the CrossPlane committed to our **main** branch, grouped by the originating ARM service.
```
Comment (matthchr) [inline hack/crossplane/apis/resources.md:4]: minor: This reads weird. These are the Crossplane _resources_?

Also it's `Crossplane` not `CrossPlane` I believe.

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go [hunk 2]
```diff
@@ -229,3 +240,40 @@ func (report *ResourceVersionsReport) generateSupportedFrom(typeName astmodel.Ty
...
+		buffer.WriteString("\n")
+		if index >= 2 && strings.HasPrefix(line, "---") {
+			break
+		}
+	}
+
+	return buffer.String()
+}
+
+// defaultFrontMatter returns the default frontmatter for the report if no existing file is present
```
Comment (matthchr) [inline v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:272]: If no file is present? Or if no front matter is present?

**Review by matthchr — APPROVED** [review]

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go (hunk 2 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:272]: Both. Comment updated.

File: hack/crossplane/apis/resources.md (hunk 1 above)
Comment (theunrepentantgeek) [reply hack/crossplane/apis/resources.md]: Amended wording and fixed spelling.

**Pull request merged**